### PR TITLE
Add XML comments to MQTT transport classes.

### PR DESCRIPTION
**Pull request by azabbasi**
Most of these xml comments are picked from DotNetty library itself. the rest of them are either a not-really-helpful comments or my best attempt at giving them meaningful descriptions.
Feel free to make suggestions if you have any context on what different methods and properties are.

### CODE REVIEW

File: iothub/device/src/Transport/Mqtt/ClientWebSocketChannelConfig.cs [hunk 1]
```diff
@@ -102,20 +118,44 @@ public bool SetOption<T>(ChannelOption<T> option, T value)
...
+        /// </summary>
         public int WriteSpinCount { get; set; }
 
+        /// <summary>
+        /// Thread-safe interface for allocating IByteBuffer
+        /// </summary>
         public IByteBufferAllocator Allocator { get; set; }
 
+        /// <summary>
+        /// Allocates a new receive buffer whose capacity is probably large enough to read
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Mqtt/ClientWebSocketChannelConfig.cs:137]: nit: "probably"?

**Review by abhipsaMisra — APPROVED** [review]
Unfortunately I don't have any additional context to add to these descriptions. 
Do you think it is worth reaching out to Max and getting some clarification from him? (afaik dotnetty documentation leaves a lot to be desired).

File: iothub/device/src/Transport/Mqtt/ClientWebSocketChannelConfig.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/device/src/Transport/Mqtt/ClientWebSocketChannelConfig.cs:137]: straight from dotnetty docs :)

Comment (abhipsaMisra) [reply iothub/device/src/Transport/Mqtt/ClientWebSocketChannelConfig.cs:137]: weird, right? 🙄

**Review by vinagesh — APPROVED** [review]

**Review by barustum — APPROVED** [review]

**Pull request merged**